### DISCUSSION TITLE
Changing channel for ProHD  Video Source=0 (Tuner)

### DISCUSSION THREAD

**Discussion by bradut** [GerryDazoo/SlingerPlayer#11] [General]
Thank you Gerry for making **SlingerPlayer** !

I've just started using it and it works great.

I have one PRO-HD here in Canada, which controls the TV box via IR dongles, and streaming and changing channels with the SlingerPlayer work like a charm.

My other two PRO-HDs are located overseas and get analogue signal via TV Cable by using the **tuner** function.
They _do not use IR to change the channels_, therefore in the unified_config.ini file I've set: `VideoSource = 0 `.

Unfortunately _I cannot change the channels_ on these boxes from Player

Here are my settings
```
[mySling]
sbtype="Solo/Pro/ProHD"
password=....    
ipaddress=....
;finderid=
port=...
Resolution=12
FrameRate=25
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=0
RemoteLock=yes
StartChannel=0
;Remote=solo_remote.txt
```

When I try to go to channel with the index 1 by clicking on "1", the server displays:
```
Remote Control Connected
Remote 1 9
mySling Got Streamer Control Message IR
IR [b'\t192.168.1.10']
04/14/2023, 23:49:42.03  mySling Sending IR keycode 9 0 for 192.168.1.10
sending key 9 0
.04/14/2023,23:49:51 mySling 1 Clients.192.168.1.10:13418
.........04/14/2023,23:51:25 mySling 1 Clients.192.168.1.10:13418
.........04/14/2023,23:52:58 mySling 1 Clients.192.168.1.10:13418
.........04/14/2023,23:54:31 mySling 1 Clients.192.168.1.10:13418
....
```  

If I try to change channel by using the textbox besides the "Channel" button, I see pretty much the same:
```
Remote Control Connected
GET Remote/mySling
..04/15/2023, 00:10:28.8   RemoteControl connection from ('192.168.1.10', 14024)
Remote Control Connected
Sending Channel Digits 1
mySling Got Streamer Control Message IR
IR [b'\t192.168.1.10']
04/15/2023, 00:10:28.901  mySling Sending IR keycode 9 0 for 192.168.1.10
sending key 9 0
....04/15/2023,00:11:09 mySling 1 Clients.192.168.1.10:13839
......
``` 

Please note that, for analogue signal, the channel numbers are assigned by Slingbox in the ascending order of their frequency.
These numbers are actually _indexes_ from 0 to 50 like this: 
```
--------------------------
Channel	        Frequency
--------------------------
 000              111.75
 001              118.75
```
...

**How can I address this issue?**

**Comment by GerryDazoo** [comment]
You need to use the ProHD remote config file to change channels. Your sending IR codes not commands to the internal tuner.

**Reply by bradut** [reply]
Thank you Gerry, this did solve most of the problem !
- I can use `Ch+`, `Ch-` and `Last`.
- I still cannot go to a channel by number: When I input the channel number, and click on the "Channel" button, it still sends the IR code to the tuner:

```
Remote Control Connected
Sending Channel Digits 2
mySling Got Streamer Control Message IR
IR [b'\n192.168.1.10']
04/15/2023, 11:07:21.450  mySling Sending IR keycode 10 0 for 192.168.1.10
sending key 10 0
cmd: 0x87 err: 0xa 120
.........04/15/2023,11:08:47 mySling 1 Clients.192.168.1.10:21278
...

```

<img src="https://user-images.githubusercontent.com/2496596/232245707-1e8a037f-bb83-46bb-a0b8-690501d24907.png" width="30%" height="30%" alt="Pro-HD Change Channel">




The config section:
```
...
RemoteLock=yes
StartChannel=0
Remote=ProHD_Tuner.txt
```

**Reply by GerryDazoo** [reply]
somewhere in the updates it has this info, You need to add a decimal point to the channel number so the code can tell the  difference between IR and internal tuner. So in the US you can say 4.0  or 9.1 for subchannels. In the UK just put in the channel number with a decimal  101.

**Reply by bradut** [reply]
Adding a decimal solved the issue,  **Thank you** ! 

<img src="https://user-images.githubusercontent.com/2496596/232252117-1fd68fda-11f3-4993-8d69-4ae011086d2d.png" width="30%" height="30%" alt="Pro-HD Change Channel">

<img src="https://user-images.githubusercontent.com/2496596/232252163-7d07a6e3-6c50-4959-9e7f-f5fbd6d5ea11.png" width="30%" height="30%" alt="Pro-HD Change Channel">
